app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/Button.kt

```diff
@@ -0,0 +1,366 @@
+  content: @Composable RowScope.() -> Unit,
+) {
+  @Suppress("NAME_SHADOWING")
+  val interactionSource = interactionSource ?: remember { MutableInteractionSource() }
+  val buttonColors = buttonStyle.style.buttonColors
+  val containerColor = buttonColors.containerColor(enabled)
+  val contentColor = buttonColors.contentColor(enabled)
+  val isHovered by interactionSource.collectIsHoveredAsState()
+  val hoverColor = buttonColors.hoverColor(isHovered)
+  val color by animateColorAsState(
+    if (isHovered) {
+      hoverColor.compositeOver(containerColor)
+    } else {
+      containerColor
+    },
+  )
+  Surface(
+    onClick = onClick,
+    modifier = modifier.semantics { role = Role.Button },
+    enabled = enabled,
+    shape = buttonSize.size.shape,
+    color = color,
+    contentColor = contentColor,
+    interactionSource = interactionSource,
+  ) {
+    ProvideTextStyle(buttonSize.size.textStyle) {
+      Row(
+        modifier = Modifier.padding(buttonSize.size.contentPadding),
+        horizontalArrangement = Arrangement.Center,
+        verticalAlignment = Alignment.CenterVertically,
+        content = content,
+      )
+    }
+  }
+}
+
+object ButtonDefaults {
+  internal val buttonStyle: ButtonStyle = ButtonStyle.Primary
+  internal val buttonSize: ButtonSize = ButtonSize.Large
+
+  enum class ButtonStyle {
+    Primary,
+    PrimaryAlt,
+    Secondary,
+    SecondaryAlt,
+    Ghost,
+  }
+
+  enum class ButtonSize {
+    Large,
+    Medium,
+    Small,
+    Mini,
+  }
+}
+
+private val ButtonDefaults.ButtonStyle.style: Style
+  get() = when (this) {
+    ButtonDefaults.ButtonStyle.Primary -> Style.Primary
+    ButtonDefaults.ButtonStyle.PrimaryAlt -> Style.PrimaryAlt
+    ButtonDefaults.ButtonStyle.Secondary -> Style.Secondary
+    ButtonDefaults.ButtonStyle.SecondaryAlt -> Style.SecondaryAlt
+    ButtonDefaults.ButtonStyle.Ghost -> Style.Ghost
+  }
+
+private val ButtonDefaults.ButtonSize.size: Size
+  get() = when (this) {
+    ButtonDefaults.ButtonSize.Large -> Size.Large
+    ButtonDefaults.ButtonSize.Medium -> Size.Medium
+    ButtonDefaults.ButtonSize.Small -> Size.Small
+    ButtonDefaults.ButtonSize.Mini -> Size.Mini
+  }
+
+@Immutable
+private data class ButtonColors(
+  val containerColor: Color,
+  val contentColor: Color,
+  val disabledContainerColor: Color,
+  val disabledContentColor: Color,
+  val hoverContainerColor: Color,
+  val hoverContentColor: Color,
+) {
+  @Stable
+  internal fun containerColor(enabled: Boolean): Color = when {
+    !enabled -> disabledContainerColor
+    else -> containerColor
+  }
+
+  @Stable
+  internal fun contentColor(enabled: Boolean): Color = when {
+    !enabled -> disabledContentColor
+    else -> contentColor
+  }
+
+  internal fun hoverColor(isHovered: Boolean): Color = when {
+    isHovered -> hoverContainerColor
+    else -> Color.Unspecified
+  }
+}
+
+private sealed interface Size {
+  val contentPadding: PaddingValues
+
+  @get:Composable
```

Comment:
that a new annotation for me

Reply:
Yep, it's the Kotlin way to say that the getter has to be annotated with the annotation that you provide after the get: part. It's interesting indeed, if you tried to put the annotation normally on the val it just doesn't work here 😊